packages/yoastseo/src/scoring/assessments/readability/SentenceBeginningsAssessment.js

```diff
@@ -36,127 +35,123 @@ export default class SentenceBeginningsAssessment extends Assessment {
-						groupedSentenceBeginnings.total,
-						"wordpress-seo"
-					),
-					this._config.urlTitle,
-					"</a>",
-					groupedSentenceBeginnings.lowestCount,
+			assessmentResult.setScore( 3 );
+			assessmentResult.setHasMarks( true );
+			// eslint-disable-next-line @wordpress/valid-sprintf -- The plural uses one extra argument.
+			assessmentResult.setText( sprintf(
+				/* translators: %1$s and %5$s expand to a link on yoast.com, %2$s expands to the anchor end tag,
+				%3$d expands to the number of consecutive sentences starting with the same word,
+				%4$d expands to the number of instances where 3 or more consecutive sentences start with the same word. */
+				_n(
+					"%1$sConsecutive sentences%2$s: The text contains %3$d consecutive sentences starting with the same word. %5$sTry to mix things up%2$s!",
+					"%1$sConsecutive sentences%2$s: The text contains %4$d instances where %3$d or more consecutive sentences start with the same word. %5$sTry to mix things up%2$s!",
 					groupedSentenceBeginnings.total,
-					this._config.urlCallToAction
+					"wordpress-seo"
 				),
-			};
-		}
-		return {
-			score: 9,
-			hasMarks: false,
-			text: sprintf(
+				this._config.urlTitle,
+				"</a>",
+				groupedSentenceBeginnings.lowestCount,
+				groupedSentenceBeginnings.total,
+				this._config.urlCallToAction
+			) );
+		} else {
+			assessmentResult.setScore( 9 );
+			assessmentResult.setHasMarks( false );
+			assessmentResult.setText( sprintf(
 				/* translators:  %1$s expands to a link on yoast.com, %2$s expands to the anchor end tag */
 				__(
 					"%1$sConsecutive sentences%2$s: There is enough variety in your sentences. That's great!",
 					"wordpress-seo"
 				),
 				this._config.urlTitle,
 				"</a>"
-			),
-		};
+			) );
+		}
+
+		return assessmentResult;
 	}
 
 	/**
 	 * Marks all consecutive sentences with the same beginnings.
 	 *
-	 * @param {object} paper        The paper to use for the assessment.
-	 * @param {object} researcher   The researcher used for calling research.
+	 * @param {Paper} paper             The paper to use for the assessment.
+	 * @param {Researcher} researcher   The researcher used for calling research.
 	 *
-	 * @returns {object} All marked sentences.
+	 * @returns {Mark[]} All marked sentences.
 	 */
 	getMarks( paper, researcher ) {
 		let sentenceBeginnings = researcher.getResearch( "getSentenceBeginnings" );
-		sentenceBeginnings = filter( sentenceBeginnings, function( sentenceBeginning ) {
-			return sentenceBeginning.count > maximumConsecutiveDuplicates;
-		} );
+		sentenceBeginnings = sentenceBeginnings.filter( sentenceBeginning => sentenceBeginning.count > MAX_SAME_BEGINNINGS );
+		const sentences = sentenceBeginnings.flatMap( sentenceBeginning => sentenceBeginning.sentences );
+		return sentences.map( sentence => {
+			const firstToken = sentence.tokens.find( ( { text } ) => text !== " " );
+			const lastToken = sentence.tokens.slice().reverse().find( ( { text } ) => text !== " " );
```

Comment:
I remember using this combination of `slice` and `reverse` to not mutate the original array, and  [`.toReserved()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/toReversed)  was not supported then. But now since we support newer JS in the package, maybe we can use `.toReserved()` instead of `.slice().reverse()` ?

Reply:
`findLast` is even better! 🤩